File: DMOJ/CCC/2019/ccc19j3.cpp
// CCC '19 J3 - Cold Compress
#include <bits/stdc++.h>
using namespace std;

#define endl "\n"
#define LOOP(n) for (int i = 0; i < n; i++)


int main() {
    ios::sync_with_stdio(0); cin.tie(0); cout.tie(0);

    int N; cin >> N;
    LOOP(N) {
        string S; cin >> S; int N1 = S.size();
        string res = "";
        char hold = S[0]; int ct = 1;
        for (int j = 1; j < N1; j++) {
            char c = S[i];
            if (c == hold) ct++;
            else if (hold && c != hold) {
                res = res + to_string(ct) + " " + hold + " ";
                hold = c; ct = 1;
            }
            if (j == N1) {
                res = res + to_string(ct) + " " + hold + " ";
            }
        }
        cout << res << endl;
    }
    

    return 0;
}

File: Canadian Computing Competition/CCC/2019/ccc19j3.py
# CCC '19 J3 - Cold Compress
ipt = [input() for i in range(int(input()))]

for i in ipt:
    # input = "+++===!!!!!"
    # output = "3 + 3 = 5 !"
    res = ''
    hold,ct = '',0
    for k,j in enumerate(i):
        if not hold:
            hold = j
            ct = 1
        elif hold and j != hold:
            res += f'{ct} {hold} '
            hold = j
            ct = 1
        elif j == hold:
            ct += 1
        if k == len(i)-1:
            res += f'{ct} {hold}'
    print(res.strip())

